File: p2119.py
class Solution(object):
    def isSameAfterReversals(self, num):
        """
        :type num: int
        :rtype: bool
        """

        # str1 = str(num)
        # rev_str1 = str1[::-1]
        # num1 = int(rev_str1)

        # str2 = str(num1)
        # rev_str2 = str2[::-1]
        # num2 = int(rev_str2)

        # if num2 == num:
        #     return True
        # else:
        #     return False

        if num == 0:
            return True
        elif num % 10 == 0:
            return False
        
        return True

File: p2119.java
class Solution {
    public boolean isSameAfterReversals(int num) {
        // // My solution
        // String numStr1 = Integer.toString(num);
        // String revStr1 = new StringBuilder(numStr1).reverse().toString();
        // int num1 = Integer.parseInt(revStr1);

        // String numStr2 = Integer.toString(num1);
        // String revStr2 = new StringBuilder(numStr2).reverse().toString();
        // int num2 = Integer.parseInt(revStr2);

        // if (num == num2){
        //     return true;
        // } else{
        //     return false;
        // }

        
        // General approach
        if (num == 0){
            return true;
        } else if (num % 10 == 0){
            return false;
        }

        return true;
    }
}